arch/arm/src/stm32h7/Make.defs

```diff
@@ -137,11 +137,6 @@ CHIP_CSRCS += stm32_adc.c
 endif
 
 ifeq ($(CONFIG_STM32H7_BBSRAM),y)
-ifeq ($(CONFIG_ARMV7M_DCACHE),y)
```

Comment:
The MPU has to be used to change the caching on BBSRAM. Did  you test that configuration?

Reply:
This was really added twice; the mpu is already included on line 95:
ifeq ($(CONFIG_ARM_MPU),y)
CMN_CSRCS += arm_mpu.c
endif

This is exactly my configuration: CONFIG_STM32H7_BBSRAM=y, CONFIG_ARMV7M_DCACHE=y, CONFIG_ARM_MPU=y.

This configuration would pull in the arm_mpu.c twice, which causes warnings/errors in compilation.

Reply:
Ahh it was broken here https://github.com/jlaitine/incubator-nuttx/commit/acb16e087a664b185438453d6dea38b7695b9a5b#diff-1ba9f72813a640243422a15396ce3038R137 

Does it error now with CONFIG_ARMV7M_DCACHE off?

I think the Kconfig should select the MPU if CONFIG_STM32H7_BBSRAM & CONFIG_ARMV7M_DCACHE